### PR TITLE
#1231 Apply template server

### PR DESCRIPTION
**Pull request by Yuseon-Han**
Letting 'nugu-ux-kit' provide TemplateView
- Add TemplateView interface to support Template displaying
- Add Web and Native View object which is implemented TemplateView interface
- Remove native views in Sample App
- Use TemplateView instead previous native views in Sample App

PMS#OPENSDK-53

### CODE REVIEW

**Review by shshshl09 — APPROVED** [review]

File: nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/view/media/PlayerCommand.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package com.skt.nugu.sdk.platform.android.ux.template.view.media
+
+enum class PlayerCommand(val command: String) {
```
Comment (tommy5oo) [inline nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/view/media/PlayerCommand.kt:3]: Proguard issue?

File: nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/model/TemplateContext.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.skt.nugu.sdk.platform.android.ux.template.model
+
+data class TemplateContext(
```
Comment (tommy5oo) [inline nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/model/TemplateContext.kt:3]: Proguard issue?

File: nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/view/media/PlayerCommand.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package com.skt.nugu.sdk.platform.android.ux.template.view.media
```
Comment (tommy5oo) [inline nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/view/media/PlayerCommand.kt:1]: Missing licence

File: nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/view/media/PlayerCommand.kt (hunk 1 above)
Comment (Yuseon-Han) [reply nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/view/media/PlayerCommand.kt:21]: fixed on new commit.

File: nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/model/TemplateContext.kt (hunk 2 above)
Comment (Yuseon-Han) [reply nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/model/TemplateContext.kt:21]: fixed on new commit.

File: nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/view/media/PlayerCommand.kt (hunk 3 above)
Comment (Yuseon-Han) [reply nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/view/media/PlayerCommand.kt:16]: fixed on new commit.

File: nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/model/TemplateContext.kt (hunk 2 above)
Comment (tommy5oo) [reply nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/model/TemplateContext.kt:21]: Thanks!

File: nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/view/media/PlayerCommand.kt (hunk 3 above)
Comment (tommy5oo) [reply nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/view/media/PlayerCommand.kt:16]: Thanks!

File: nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/view/media/PlayerCommand.kt (hunk 1 above)
Comment (tommy5oo) [reply nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/view/media/PlayerCommand.kt:21]: Thanks!

File: app/src/main/java/com/skt/nugu/sampleapp/template/TemplateFragment.kt [hunk 4]
```diff
@@ -13,305 +13,192 @@
...
+    private fun initView() {
+        templateView = TemplateView.createView(getTemplateType(), requireContext().applicationContext)
 
-                view.setOnClickListener {
-                    ClientManager.getClient().localStopTTS()
+        with(templateView!!) {
+            (this as? WebView)?.webViewClient = object : WebViewClient() {
+                override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
+                    if (request?.url.toString() == "nugu://home") {
+                        (activity as? MainActivity)?.clearAllTemplates()
```
Comment (tommy5oo) [inline app/src/main/java/com/skt/nugu/sampleapp/template/TemplateFragment.kt:108]: Unresolved reference: MainActivity

File: nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/webview/TemplateWebView.kt [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+        settings.apply {
+            defaultTextEncodingName = "UTF-8"
+            javaScriptEnabled = true
+            domStorageEnabled = true
+            builtInZoomControls = false
+            savePassword = false
+            saveFormData = false
+            setSupportZoom(false)
+            builtInZoomControls = false
+            useWideViewPort = false
```
Comment (tommy5oo) [inline nugu-ux-kit/src/main/java/com/skt/nugu/sdk/platform/android/ux/template/webview/TemplateWebView.kt:81]: Shouldn't useWideViewPort be set to true?

**Pull request closed without merging**